samples/azure/azure-container-apps-dab-starter.ps1

```diff
@@ -0,0 +1,744 @@
+    param([string]$Prefix, [string]$Type)
+    $random = -join ((48..57) + (97..122) | Get-Random -Count 6 | ForEach-Object {[char]$_})
+    return "$Prefix-$Type-$random".ToLower()
+}
+
+function Update-DabConfigFile {
+    param(
+        [string]$SourceConfigPath,
+        [string]$ConnectionString,
+        [string]$OutputPath
+    )
+    
+    # Read the source config file
+    $configContent = Get-Content -Path $SourceConfigPath -Raw
+    
+    # Parse JSON
+    $config = $configContent | ConvertFrom-Json
+    
+    # Update the connection string
+    $config.'data-source'.'connection-string' = $ConnectionString
+    
+    # Ensure production mode for Container Apps
+    if ($config.runtime.host.mode -eq "development") {
+        $config.runtime.host.mode = "production"
+    }
+    
+    # Convert back to JSON and save
+    $configContent = $config | ConvertTo-Json -Depth 100
+    $configContent | Set-Content -Path $OutputPath -Encoding UTF8
+    
+    Write-Success "Updated DAB config file: $OutputPath"
+}
+
+# ============================================================================
+# MAIN SCRIPT
+# ============================================================================
+
+Write-Host "`n" -ForegroundColor Blue
+Write-Host "==========================================================================" -ForegroundColor Blue
+Write-Host "  Data API Builder - Azure Container Apps Deployment" -ForegroundColor Blue
+Write-Host "==========================================================================" -ForegroundColor Blue
+Write-Host ""
+
+# Check prerequisites
+Test-Prerequisites
+
+# Find repo root
+$repoPath = $PSScriptRoot
+while ($repoPath -and -not (Test-Path (Join-Path $repoPath "Dockerfile"))) {
+    $repoPath = Split-Path $repoPath -Parent
+}
+
+if (-not $repoPath -or -not (Test-Path (Join-Path $repoPath "Dockerfile"))) {
+    Write-ErrorMessage "Could not find data-api-builder repository root (looking for Dockerfile)"
+    exit 1
+}
+
+$sqlScriptPath = Join-Path $repoPath "src\Service.Tests\DatabaseSchema-MsSql.sql"
+if (-not (Test-Path $sqlScriptPath)) {
+    Write-ErrorMessage "Could not find SQL script at: $sqlScriptPath"
+    exit 1
+}
+
+# Set default DAB config file path if not provided
+if (-not $DabConfigFile) {
+    $DabConfigFile = Join-Path $repoPath "src\Service.Tests\dab-config.MsSql.json"
+}
+
+if (-not (Test-Path $DabConfigFile)) {
+    Write-ErrorMessage "DAB config file not found at: $DabConfigFile"
+    exit 1
+}
+
+Write-Success "Repository root: $repoPath"
+Write-Success "Using DAB config: $DabConfigFile"
+
+# Generate or validate parameters
+if (-not $ResourcePrefix) {
+    $ResourcePrefix = "dab$(Get-Random -Maximum 9999)"
+    Write-Host "Generated resource prefix: $ResourcePrefix" -ForegroundColor Yellow
+}
+
+if (-not $ResourceGroup) {
+    $ResourceGroup = "rg-$ResourcePrefix"
+    Write-Host "Generated resource group: $ResourceGroup" -ForegroundColor Yellow
+}
+
+# Set subscription
```

Comment:
Since this is a recommended sample, we should steer away from using sql password. Can we use Managed Identity to login to the SQL that we create? So, the ACA hosting DAB that gets deployed should use Managed Identity to login to the deployed SQL db. You can fix it as part of a new PR.